aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -72,9 +72,34 @@ void scatter_shape_check(
     }
   }
   TORCH_CHECK(!is_wrong_shape,
-    "Expected index [", index.sizes(), "]",
-    " to be smaller size than src [", src.sizes(), "]",
-    " and to be smaller than self [", self.sizes(), "]",
+    "Expected index ", index.sizes(),
+    " to be smaller size than src ", src.sizes(),
+    " and to be smaller than self ", self.sizes(),
+    " apart from dimension ", dim
+  );
+}
+
+// Used for `scatter` when source is Scalar
+// Tests:
+//  1. index.size(d) <= self.size(d) for all d != dim
+void scatter_shape_check(
```

Comment:
`scatter_shape_check` overloads are confusing and repetitive, can they be handled in the same function with an optional `src` argument?